src/main/java/seedu/address/model/pet/comparator/PetDropOffTimeComparator.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.pet.comparator;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Comparator;
+import java.util.Optional;
+
+import seedu.address.model.attendance.AttendanceEntry;
+import seedu.address.model.pet.Pet;
+
+/**
+ * Comparator class to compare pets based on pet's Drop Off Time.
+ */
+public class PetDropOffTimeComparator implements Comparator<Pet> {
+    private LocalDate today = LocalDate.now();
+
+    /**
+     * Compares pet objects based on their drop off time attribute.
+     * @param first Pet to be compared with.
+     * @param second Pet to be compared with.
+     * @return Value signifying in the difference between the comparing attribute.
+     */
+    @Override
+    public int compare(Pet first, Pet second) {
+        Optional<AttendanceEntry> firstAttendanceEntry = first.getAttendanceHashMap().getAttendance(today);
+        Optional<AttendanceEntry> secondAttendanceEntry = second.getAttendanceHashMap().getAttendance(today);
+
+        if (firstAttendanceEntry.isPresent() && secondAttendanceEntry.isPresent()) {
+            return compareAttendanceEntries(firstAttendanceEntry, secondAttendanceEntry);
```

Comment:
It is better to pass in the actual entry (from calling `Optional#get()`), rather than pass Optionals as a method parameter!

Same for `PetPickUpTimeComparator`.